nav2_waypoint_follower/src/waypoint_follower.cpp

```diff
@@ -126,7 +126,7 @@ WaypointFollower::followWaypoints()
     static_cast<int>(goal->poses.size()));
 
   rclcpp::Rate r(loop_rate_);
-  uint goal_index = 0;
+  uint32_t goal_index = 0;
```

Comment:
Does `uint` cause windows problems?

Reply:
`uint` seems to be a [typedef from Glibc](https://gcc.gnu.org/bugzilla/show_bug.cgi?id=59945) and it is not a C\C++ language defined data type. So I decided to change it to a more portable data type, and the reason I chose `uint32_t` is to be aligned with the data type used by [`feedback->current_waypoint`](https://github.com/ros-planning/navigation2/blob/81805709fd32cc2619ae39c2fbd6c3facaf4f01f/nav2_waypoint_follower/src/waypoint_follower.cpp#L164).

`unsigned int` works too on Windows, which is a [4-byte data type](https://docs.microsoft.com/en-us/cpp/cpp/data-type-ranges?view=vs-2019), so it should not be a problem.

Reply:
OK got it. At some point, not now, it may be useful to see if there's something we can do to add Windows stuff to CI to catch these kind of issues. Either as a docker image or however the Windows workflow generally works.

Reply:
I think there are some Windows CI support from ROS2 CI buildfarm. Perhaps we just need to figure out how to plug this repository there?

cc @brawner

Reply:
I'm less concerned with this particular repo and more with the ecosystem in general. There should exist a general solution to support Windows the way we have for Linux if you want the community to support Windows. E.g. build and testing tools to verify that it works in Windows and guides to fixing issues that might appear since they're uniquely Windows for the most part.

Reply:
Are there plans to add navigation2 to the [ros2.repos?](github.com/ros2/ros2) Because that is what is currently tested on ci.ros2.org. Otherwise, I've seen a lot of repos use Appveyor for their Windows CI?

Reply:
There's no reason to expect to, no. Navigation2 isn't a generic tool used by all users (manipulators, cars, mobile, drones, submarines, etc). That repos file is general tools.

Reply:
@brawner I didn't know the `ci.ros2.org` only for essential packages. Sorry for any confusion I made.

@SteveMacenski Thanks for the feedback that the Windows specific knowledge is less visible to ROS community. I and my team will help improve this. Regarding for using public CI like GitHub workflow, I'd recommend we can begin with the GitHub Actions curated by [ROS 2 Tooling Working Group](https://github.com/ros-tooling/community). Windows and Linux are supported there. Let me know what you think.

Reply:
Is there a windows environment for that?

Reply:
@SteveMacenski Yes, the `action-ros-ci` and `setup-ros` both work with Windows environment on GitHub workflow (which has Linux, MacOSX, and Window environments.) If you think GitHub workflow is a good direction to host the Windows-side CI, I can spend some time investigating that (in a separate pull request).

Reply:
Got it. At minimum, something that just builds relative to {master, released binaries of XYZ version} would be great. Even build failures due to Windows-ing is a massive improvement.